### PR TITLE
WIP: Exp/api fetch

### PR DESCRIPTION
**Pull request by KenLSM**
Not intended for merge.

### CODE REVIEW

File: frontend/src/features/admin-form/preview/PreviewFormProvider.tsx [hunk 1]
```diff
@@ -130,28 +130,60 @@ export const PreviewFormProvider = ({
...
+                onSuccess: ({ submissionId }) =>
+                  setSubmissionData({
+                    id: submissionId,
+                    timestamp: Date.now(),
+                  }),
+              },
+            )
+          }
+          const submitStorageModeFormMutationWithFetchStrategy = () => {
+            return submitStorageModeFormMutation.mutateAsync(
```
Comment (KenLSM) [inline frontend/src/features/admin-form/preview/PreviewFormProvider.tsx:153]: Not yet implemented. 

TODO: create a new mutation that uses fetch

File: frontend/src/features/admin-form/preview/PreviewFormProvider.tsx [hunk 2]
```diff
@@ -130,28 +130,60 @@ export const PreviewFormProvider = ({
...
+          const submissionStrategies = [
+            submitStorageModeFormMutationWithAxiosStrategy,
+            submitStorageModeFormMutationWithFetchStrategy,
+          ]
+          let lastError
+          // iterate through all submission strategies until one passes
+          for (const strategy of submissionStrategies) {
+            try {
+              await strategy()
+              return
```
Comment (KenLSM) [inline frontend/src/features/admin-form/preview/PreviewFormProvider.tsx:178]: return immediately if the strategy works

File: frontend/src/features/public-form/utils/axiosDebugFlow.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    meta: {
+      action: 'handleSubmitForm',
+      envAxiosFetchSuccess: env.ok, // returns true if the response returned successfully
+    },
+  })
+}
+export const axiosDebugFlow = async (error: AxiosError) => {
+  if (error.message.match(/Network Error/i)) {
+    getClientEnvWithFetch()
+    getClientEnvWithAxios()
```
Comment (KenLSM) [inline frontend/src/features/public-form/utils/axiosDebugFlow.tsx:30]: Thinking of double checking that its not something wrong with the payload, and calling a trivial API with both `fetch` and `axios` should help scope the problem down to `axios`.

File: frontend/src/services/ApiService.ts [hunk 4]
```diff
@@ -43,8 +43,10 @@ export const transformAxiosError = (error: Error): ApiError => {
         meta: {
           action: 'transformAxiosError',
           error: {
+            code: error?.code,
             message: error?.message,
             stack: error?.stack,
+            dump: error?.toJSON(),
```
Comment (KenLSM) [inline frontend/src/services/ApiService.ts:49]: Adding a few more error logs incase it might yield more insights

**Comment by KenLSM** [comment]
Hi @wanlingt, I might need to pass this back to you. I've left some comments in the PR to hopefully translate what I was thinking.

**Comment by wanlingt** [comment]
[Datadog logs](https://app.datadoghq.com/logs?query=service%3Aformsg%20%20%20%22network%20error%22&additional_filters=%5B%7B%7D%5D&cols=%40network.client.ip%2C%40http.useragent_details.browser.family%2C%40http.useragent_details.browser.major&index=%2A&messageDisplay=inline&saved-view-id=1483813&stream_sort=time%2Cdesc&viz=stream&from_ts=1675934493611&to_ts=1676539293611&live=true)

**Comment by KenLSM** [comment]
Closing in favour of #5972

**Pull request closed without merging**